### ISSUE TITLE
Forwarding command line options to simulation server

### ISSUE THREAD

**Issue by lochel** [OpenModelica/OpenModelica#7170]
### Description
The command line options specified in OMEdit are not forwarded to the simulation server. Do we want to split it up into options for modelling and another one for simulation? I think that would be reasonable - but maybe slightly confusing. But those options are only meant to be used by experts anyways.

### Screenshots
![image](https://user-images.githubusercontent.com/8457302/108179662-28039d00-7106-11eb-9d23-8b15dab5ce57.png)

**Comment by adeas31** [comment]
It seems the simulation server also needs to be extended to support this. We should also handle logging as well, right?

**Comment by lochel** [comment]
Yes, that's correct. I am already working on the changes for the simulation server. I will also include the logging.

What do you think about splitting the command line options into two sets of options, one for OMEdit and one for the server?

**Comment by adeas31** [comment]
> What do you think about splitting the command line options into two sets of options, one for OMEdit and one for the server?

I am not sure if this really helps. For now we can have just one set of options.

**Comment by lochel** [comment]
Using the options to set the work directory or the temp directory might cause problems if we are not separating them.

**Comment by adeas31** [comment]
How? I don't use the temp argument of simulation server anyway.

**Comment by lochel** [comment]
Because we would use the same temp directory for modelling and simulation - two independent processes. I guess it would work fine, but it could result in name clashes.

**Comment by lochel** [comment]
@adeas31 Now you can forward the command line options (`--option`) and the logging level (`--logLevel`).

```log
> OMSimulatorPython3 build/share/OMSimulator/scripts/OMSimulatorServer.py -h
usage: OMSimulatorServer.py [-h] [--endpoint-pub ENDPOINT_PUB] [--endpoint-rep ENDPOINT_REP] --model MODEL
                            [--logLevel LOGLEVEL] [--option OPTION] [--result-file RESULT_FILE] [--temp TEMP]

OMS-SERVER

optional arguments:
  -h, --help            show this help message and exit
  --endpoint-pub ENDPOINT_PUB
                        define the endpoint for the pub/sub communication
  --endpoint-rep ENDPOINT_REP
                        define the endpoint for the req/rep communication
  --model MODEL         define the model to simulate
  --logLevel LOGLEVEL   defines the logging level
  --option OPTION       defines optional command line options
  --result-file RESULT_FILE
                        defines whether and if so to which file the results will be written
  --temp TEMP           defines the temp directory
```

**Comment by adeas31** [comment]
Nice! Is the options argument a space separated list?

**Comment by lochel** [comment]
No, it is just a string like, exactly as in the textbox in OMEdit.

**Comment by lochel** [comment]
I was a bit too fast with merging your pull request. It unfortunately doesn't work when using two flags. The python package `argparse` doesn't play well with spaces...

**Comment by lochel** [comment]
A workaround is to use comma instead of space to separate the flags and to replace comma with space in the `OMSimulatorServer.py`. But that is not nice 😒 

```diff
diff --git a/src/OMSimulatorServer/OMSimulatorServer.py b/src/OMSimulatorServer/OMSimulatorServer.py
index 073dd20..c0bd809 100755
--- a/src/OMSimulatorServer/OMSimulatorServer.py
+++ b/src/OMSimulatorServer/OMSimulatorServer.py
@@ -86,7 +86,7 @@ def _main():
   else:
     socket_sub = None

-  oms.setCommandLineOption(args.option)
+  oms.setCommandLineOption(args.option.replace(',', ' '))
   oms.setLoggingLevel(args.logLevel)
   oms.setTempDirectory(args.temp)
   model = oms.importFile(args.model)
```

**Comment by lochel** [comment]
I have a better solution for it. Unfortunately, it requires both change in OMEdit and OMSimulatorPython.

**Comment by adeas31** [comment]
Looking forward to your solution. I will adapt OMEdit accordingly.

**Comment by lochel** [comment]
@adeas31 Please have a look and let me know what you think.

**Comment by adeas31** [comment]
Fixed with #7191 and OpenModelica/OMSimulator#956
